### PR TITLE
Allow omitting blank lines around one-liner definitions.

### PR DESCRIPTION
**Pull request by anntzer**
Fixes the E302 part of #231.

### CODE REVIEW

**Comment by anntzer** [comment]
kindly bumping

**Comment by sigmavirus24** [comment]
I think this looks okay. I wonder if I could impose upon @asottile for a second review though

File: pycodestyle.py [hunk 1]
```diff
@@ -354,6 +354,17 @@ def blank_lines(logical_line, blank_lines, indent_level, line_number,
...
         yield 0, "E303 too many blank lines (%d)" % blank_lines
     elif STARTSWITH_TOP_LEVEL_REGEX.match(logical_line):
+        # If this is a one-liner (i.e. the next line is not more
+        # indented), and the previous line is also not deeper
+        # (it would be better to check if the previous line is part
+        # of another def/class at the same level), don't require blank
+        # lines around this.
+        prev_line = next(
+            (line for line in lines[:line_number - 1][::-1] if line), '')
+        next_line = next((line for line in lines[line_number:] if line), '')
```
Comment (asottile) [inline pycodestyle.py]: I'd write this:

```
(Pdb) lines[line_number - 1] if line_number >= 0 else ''
'def f(): pass\n'
(Pdb) lines[line_number + 1] if line_number < len(lines) else ''
''
```

had a good head scratch trying to figure out what this was doing :laughing:

Comment (anntzer) [reply pycodestyle.py]: you're off by 1, but sure, changed that.

Comment (asottile) [reply pycodestyle.py]: ahah yeah, you get the idea ;)

**Comment by anntzer** [comment]
Let's say supporting exactly *one* blank line between one liners could be "future work"? :)

**Pull request merged**